src/Microsoft.AspNet.Mvc.ViewFeatures/ModelStateDictionaryExtensions.cs

```diff
@@ -0,0 +1,115 @@
+        {
+            modelState.AddModelError(GetExpressionText(expression), exception);
+        }
+
+        /// <summary>
+        /// Removes the specified <paramref name="expression"/> from the <see cref="ModelStateDictionary"/>.
+        /// </summary>
+        /// <typeparam name="TModel">The type of the model.</typeparam>
+        /// <param name="modelState">The <see cref="ModelStateDictionary"/> instance this method extends.</param>
+        /// <param name="expression">An expression to be evaluated against an item in the current model.</param>
+        /// <returns>
+        /// true if the element is successfully removed; otherwise, false.
+        /// This method also returns false if <paramref name="expression"/> was not found in the model-state dictionary.
+        /// </returns>
+        public static bool Remove<TModel>(this ModelStateDictionary modelState, Expression<Func<TModel, object>> expression)
+        {
+            return modelState.Remove(GetExpressionText(expression));
+        }
+
+        /// <summary>
+        /// Removes all the entries for the specified <paramref name="expression"/> from the <see cref="ModelStateDictionary"/>.
+        /// </summary>
+        /// <typeparam name="TModel">The type of the model.</typeparam>
+        /// <param name="modelState">The <see cref="ModelStateDictionary"/> instance this method extends.</param>
+        /// <param name="expression">An expression to be evaluated against an item in the current model.</param>
+        public static void RemoveAll<TModel>(this ModelStateDictionary modelState, Expression<Func<TModel, object>> expression)
+        {
+            string modelKey = GetExpressionText(expression);
+            if (string.IsNullOrEmpty(modelKey))
+            {
+                var modelMetadata = new EmptyModelMetadataProvider().GetMetadataForType(typeof(TModel));
+                var elementMetadata = modelMetadata.ElementMetadata;
+                if (elementMetadata != null)
+                {
+                    modelMetadata = elementMetadata;
+                }
+
+                foreach (var property in modelMetadata.Properties)
+                {
+                    var childKey = property.BinderModelName ?? property.PropertyName;
+                    var entries = modelState.FindKeysWithPrefix(childKey).ToArray();
+                    foreach (var entry in entries)
+                    {
+                        modelState.Remove(entry.Key);
+                    }
+                }
+            }
+            else
+            {
+                var entries = modelState.FindKeysWithPrefix(modelKey).ToArray();
+                foreach (var entry in entries)
+                {
+                    modelState.Remove(entry.Key);
+                }
+            }
+        }
+
+        private static string GetExpressionText(LambdaExpression expression)
+        {
+            var unaryExpression = expression.Body as UnaryExpression;
```

Comment:
It's all about not having to specify to two generic types then calling an extension method, so we don't have to have boilerplate code like this:
`ModelState.AddModelError<Account, *int*>(model => model.Age, "You are too young.")`
And instead can write
`ModelState.AddModelError<Account>(model => model.Age, "You are too young.")`
which on structs result in automatic cast to object, because
`Expression<Func<TModel, object>> expression` converts stuff like `model => model.Age` to `model => Convert(model.Age)` which can not be converted to proper text by `ExpressionHelper.GetExpressionText` method.